spec/requests/admin_payroll_run_payments_spec.rb

```diff
@@ -0,0 +1,72 @@
+require "rails_helper"
+
+RSpec.describe "Admin payroll run payments" do
+  let(:admin) { create(:dfe_signin_user) }
+
+  before do
+    sign_in_to_admin_with_role(DfeSignIn::User::SERVICE_OPERATOR_DFE_SIGN_IN_ROLE_CODE, admin.dfe_sign_in_id)
+  end
+
+  let(:payroll_run) { create(:payroll_run, claims_counts: {MathsAndPhysics => 1, StudentLoans => 1}) }
+  let(:payment) { payroll_run.payments.first }
+
+  describe "delete" do
+    it "shows a confirmation screen" do
+      get remove_admin_payroll_run_payment_path(
+        payroll_run_id: payroll_run.id,
+        id: payment.id
+      )
+    end
+  end
+
+  describe "destroy" do
+    it "deletes a payroll run and redirects with a message" do
+      expect {
+        delete admin_payroll_run_payment_path(
+          payroll_run_id: payroll_run.id,
+          id: payment.id
+        )
+      }.to change(payroll_run.reload.payments, :count).by(-1)
+
+      expect(payroll_run.payments).to_not include(payment)
+
+      expect(response.body).to include("Payment has been removed from payroll run")
```

Comment:
I think this spec and the feature spec should confirm the the claim references are displayed as it's an important part of the feature.